crates/fuel-core/src/graphql_api/storage/coins.rs

```diff
@@ -23,6 +41,180 @@ pub fn owner_coin_id_key(owner: &Address, coin_id: &UtxoId) -> OwnedCoinKey {
+            if message.is_retryable_message() {
+                &RETRYABLE_BYTE
+            } else {
+                &NON_RETRYABLE_BYTE
+            },
+        );
+        offset += u8::BITS as usize / 8;
+        arr[offset..offset + u64::BITS as usize / 8].copy_from_slice(&amount_bytes);
+        offset += u64::BITS as usize / 8;
+        arr[offset..offset + Nonce::LEN].copy_from_slice(nonce_bytes);
+        offset += Nonce::LEN;
+        arr[offset..].copy_from_slice(&indexation::coins_to_spend::MESSAGE_PADDING_BYTES);
+        Self(arr)
+    }
+
+    pub fn from_slice(slice: &[u8]) -> Result<Self, core::array::TryFromSliceError> {
+        Ok(Self(slice.try_into()?))
+    }
+
+    pub fn owner(&self) -> Address {
+        let address_start = 0;
+        let address_end = address_start + Address::LEN;
+        let address: [u8; Address::LEN] = self.0[address_start..address_end]
+            .try_into()
+            .expect("should have correct bytes");
+        Address::new(address)
+    }
+
+    pub fn asset_id(&self) -> AssetId {
+        let offset = Address::LEN;
+
+        let asset_id_start = offset;
+        let asset_id_end = asset_id_start + AssetId::LEN;
+        let asset_id: [u8; AssetId::LEN] = self.0[asset_id_start..asset_id_end]
+            .try_into()
+            .expect("should have correct bytes");
+        AssetId::new(asset_id)
+    }
+
+    pub fn retryable_flag(&self) -> u8 {
+        let offset = Address::LEN + AssetId::LEN;
+        self.0[offset]
+    }
+
+    #[allow(clippy::arithmetic_side_effects)]
+    pub fn amount(&self) -> u64 {
+        let offset = Address::LEN + AssetId::LEN + u8::BITS as usize / 8;
+        let amount_start = offset;
+        let amount_end = amount_start + u64::BITS as usize / 8;
+        u64::from_be_bytes(
+            self.0[amount_start..amount_end]
+                .try_into()
+                .expect("should have correct bytes"),
+        )
+    }
+
+    #[allow(clippy::arithmetic_side_effects)]
+    pub fn foreign_key_bytes(
+        &self,
+    ) -> &[u8; CoinsToSpendIndexKey::LEN
+            - Address::LEN
+            - AssetId::LEN
+            - u8::BITS as usize / 8
+            - u64::BITS as usize / 8] {
+        let offset =
+            Address::LEN + AssetId::LEN + u8::BITS as usize / 8 + u64::BITS as usize / 8;
+        self.0[offset..]
+            .try_into()
+            .expect("should have correct bytes")
+    }
+}
+
+impl TryFrom<&[u8]> for CoinsToSpendIndexKey {
+    type Error = core::array::TryFromSliceError;
+    fn try_from(slice: &[u8]) -> Result<Self, Self::Error> {
+        CoinsToSpendIndexKey::from_slice(slice)
+    }
+}
+
+impl AsRef<[u8]> for CoinsToSpendIndexKey {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl Mappable for CoinsToSpendIndex {
+    type Key = Self::OwnedKey;
+    type OwnedKey = CoinsToSpendIndexKey;
+    type Value = Self::OwnedValue;
+    type OwnedValue = u8;
```

Comment:
Why do we need 1 byte here? If you don't use it, we can just use `()` type.

Reply:
Ah, I see, it is `IndexedCoinType`. Why not to use this type here instead of `u8`? Also, in the comment above I said that maybe we could use `retryable` and `non-retryable` byte to track message/coin type.